com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.RenderGraph.cs

```diff
@@ -198,13 +198,29 @@ class TempPassData { };
                 }
 
                 // This final Gaussian pyramid can be reused by SSR, so disable it only if there is no distortion
-                if (hdCamera.frameSettings.IsEnabled(FrameSettingsField.Distortion) || hdCamera.IsSSREnabled())
-                    GenerateColorPyramid(m_RenderGraph, hdCamera, colorBuffer, currentColorPyramid, false);
+                TextureHandle distortionColorPyramid;
+                if (hdCamera.frameSettings.IsEnabled(FrameSettingsField.Distortion) && hdCamera.frameSettings.IsEnabled(FrameSettingsField.RoughDistortion))
+                {
+                    distortionColorPyramid = m_RenderGraph.CreateTexture(
```

Comment:
One thing to note here.
Before those changes, I think we'd just use a double buffered history pyramid from the hdCamera.
Here we create a new one. Since we have no other (I think) textures with mips in the pipeline, this will actually allocate more memory than before. Can't we just reuse one of the history buffers?

Reply:
discussed, not relevant in this case